gradle-plugin: `user.home` for github actions is still incorrect

https://github.com/analysis-dev/diktat/blob/755b6b7421d1a070ba598b48329d0f7468cce2cf/diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt#L195

Should use `rootDir`, not `projectDir`. Current version will yield correct results only for the root project.

The issue still affects DiKTat 1.2.1, at least on Windows. Minimal project file (using the Groovy DSL):

```groovy
buildscript {
    repositories {
        mavenCentral()
    }

    dependencies {
        classpath group: 'org.jetbrains.kotlin', name: 'kotlin-gradle-plugin', version: kotlin_version
        classpath group: 'org.cqfn.diktat', name: 'diktat-gradle-plugin', version: diktat_version
    }
}

subprojects {
    apply plugin: 'kotlin'
    apply plugin: 'org.cqfn.diktat.diktat-gradle-plugin'

    repositories {
        mavenCentral()
    }

    dependencies {
        implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
    }

    compileKotlin {
        kotlinOptions {
            jvmTarget = "1.8"
        }
    }

    compileTestKotlin {
        kotlinOptions {
            jvmTarget = "1.8"
        }
    }

    diktat {
        inputs {
            it.include("src/**/*.kt", "*.kts", "src/**/*.kts")
            it.exclude("build/**")
        }

        reporter = "sarif"
        debug = false
    }
}
```

This leads to:

```console
D:\test>gradlew diktatCheck -Pdiktat.githubActions=true                                                              
> Task :test-api:diktatCheck FAILED
Exception in thread "main" java.lang.IllegalArgumentException: this and base files have different roots: D:\test\src\main\kotlin\com\example\A.kt and C:\Users\user.
        at kotlin.io.FilesKt__UtilsKt.toRelativeString(Utils.kt:117)
        at kotlin.io.FilesKt__UtilsKt.relativeTo(Utils.kt:128)
        at com.pinterest.ktlint.reporter.sarif.SarifReporter.onLintError(SarifReporter.kt:52)
        at com.pinterest.ktlint.core.Reporter$Companion$from$1.onLintError(Reporter.kt:41)
        at com.pinterest.ktlint.internal.KtlintCommandLine.report(KtlintCommandLine.kt:332)
        at com.pinterest.ktlint.internal.KtlintCommandLine.access$report(KtlintCommandLine.kt:48)
        at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$4.invoke(KtlintCommandLine.kt:283)
        at com.pinterest.ktlint.internal.KtlintCommandLine$lintFiles$4.invoke(KtlintCommandLine.kt:283)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel(KtlintCommandLine.kt:578)
        at com.pinterest.ktlint.internal.KtlintCommandLine.parallel$default(KtlintCommandLine.kt:536)
        at com.pinterest.ktlint.internal.KtlintCommandLine.lintFiles(KtlintCommandLine.kt:283)
        at com.pinterest.ktlint.internal.KtlintCommandLine.run(KtlintCommandLine.kt:235)
        at com.pinterest.ktlint.Main.main(Main.kt:31)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test-api:diktatCheck'.
> Process 'command 'C:\Program Files\Zulu\zulu-8\bin\java.exe'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
1 actionable task: 1 executed
```

The workaround is to override `user.home`:

```groovy
import org.cqfn.diktat.plugin.gradle.DiktatJavaExecTaskBase

// ...

tasks.withType(DiktatJavaExecTaskBase).configureEach {
    systemProperty("user.home", rootDir.toString())
}
```

--- a/diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt
+++ b/diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt
@@ -170,22 +170,32 @@ open class DiktatJavaExecTaskBase @Inject constructor(
         return flag.toString()
     }
 
-    @Suppress("SAY_NO_TO_VAR")
     private fun setReporter(diktatExtension: DiktatExtension, flag: java.lang.StringBuilder) {
         val name = diktatExtension.reporter.trim()
         val validReporters = listOf("sarif", "plain", "json", "html")
-        var reporterFlag = if (name.isEmpty() || !validReporters.contains(name)) {
-            project.logger.warn("Reporter name $name was not specified or is invalid. Falling to 'plain' reporter")
-            "--reporter=plain"
-        } else {
-            "--reporter=$name"
+        val reporterFlag = when {
+            diktatExtension.githubActions -> {
+                if (diktatExtension.reporter.isNotEmpty()) {
+                    // githubActions should have higher priority than custom input
+                    project.logger.warn("`diktat.githubActions` is set to true, so custom reporter [$name] will be ignored and SARIF reporter will be used")
+                }
+                "--reporter=sarif"
+            }
+            name.isEmpty() -> {
+                project.logger.info("Reporter name was not set. Using 'plain' reporter")
+                "--reporter=plain"
+            }
+            name !in validReporters -> {
+                project.logger.warn("Reporter name is invalid (provided value: [$name]). Falling back to 'plain' reporter")
+                "--reporter=plain"
+            }
+            else -> "--reporter=$name"
         }
 
-        // githubActions should have higher priority than a custom input
-        if (diktatExtension.githubActions) {
+        val isSarifReporterActive = reporterFlag.contains("sarif")
+        if (isSarifReporterActive) {
             // need to set user.home specially for ktlint, so it will be able to put a relative path URI in SARIF
             systemProperty("user.home", project.rootDir.toString())
-            reporterFlag = "--reporter=sarif"
         }
 
         flag.append(reporterFlag)

--- a/diktat-gradle-plugin/src/test/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskTest.kt
+++ b/diktat-gradle-plugin/src/test/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskTest.kt
@@ -121,6 +121,11 @@ class DiktatJavaExecTaskTest {
             diktatConfigFile = project.file("../diktat-analysis.yml")
             githubActions = true
         }
+        val task = project.tasks.getByName(DIKTAT_CHECK_TASK) as DiktatJavaExecTaskBase
+        Assertions.assertEquals(
+            project.rootDir.toString(),
+            task.systemProperties["user.home"]
+        )
     }
 
     @Test
@@ -135,6 +140,22 @@ class DiktatJavaExecTaskTest {
             reporter = "json"
             output = "report.json"
         }
+    }
+
+    @Test
+    fun `should set system property with SARIF reporter`() {
+        assertCommandLineEquals(
+            listOf(null, "--reporter=sarif")
+        ) {
+            inputs { exclude("*") }
+            diktatConfigFile = project.file("../diktat-analysis.yml")
+            reporter = "sarif"
+        }
+        val task = project.tasks.getByName(DIKTAT_CHECK_TASK) as DiktatJavaExecTaskBase
+        Assertions.assertEquals(
+            project.rootDir.toString(),
+            task.systemProperties["user.home"]
+        )
     }
 
     @Test

